### DISCUSSION TITLE
warn for missing .env file?

### DISCUSSION THREAD

**Discussion by mathause** [encode/starlette#2101] [Ideas] [answered]
`Config(".env")` 'fails' silently if the file does not exist. Should this raise an error or give an informative warning? I think that might be helpful if you forget to create your .env on the production server.

https://github.com/encode/starlette/blob/bc90057c121da88e6d30cd06969ae4097dc6fbc2/starlette/config.py#L62

```python
from starlette.config import Config
Config(".env")
```

**Comment by Kludex** [comment, marked as answer]
PR welcome.

**Comment by timombre** [comment]
Was raising an error implemented ? Deployment pipeline has crashed when fastapi (which has starlette as a dependency) released a new version.

tests/conftest.py:9: in <module>
    from app.main import app
app/main.py:12: in <module>
    from starlette_authlib.middleware import AuthlibMiddleware as SessionMiddleware
newenv/lib/python3.11/site-packages/starlette_authlib/middleware.py:31: in <module>
    config = Config(".env")
newenv/lib/python3.11/site-packages/starlette/config.py:65: in __init__
    raise FileNotFoundError(f"Config file '{env_file}' not found.")
E   FileNotFoundError: Config file '.env' not found.

**Comment by PeterJCLaw** [comment]
It would be great if the release notes could call this out as a (breaking) change. In my project I used to use `Config('.env')` as a way to allow for an _optional_ config file (I have suitable defaults in place) and e.g: in CI tests I don't have a `.env` file. With this change I (and anyone using this behaviour) will need to manually reimplement the file check (incurring IO races along the way).

Perhaps a better solution would be to have a warning if the file is missing? Or to have classmethods on `Config` which allow for different expectations? So like `Config.with_required_file('.env')`, `Config.with_optional_file('.env')`; perhaps leaving `Config(...)` not accepting a file at all.

In case useful to others I'm now doing this:
``` python
try:
    config = Config('.env')
except FileNotFoundError:
    config = Config()
```

**Reply by bengabp** [reply]
is this implemented now ?

**Comment by Kludex** [comment]
Didn't we revert this, and use warning instead of error? or something...
